rclrs/src/node.rs

```diff
@@ -189,6 +191,26 @@ impl Node {
         Ok(client)
     }
 
+    /// Creates a [`GuardCondition`][1].
+    ///
+    /// A weak pointer to the `GuardCondition` is stored within this node.
+    /// When this node is added to a wait set (e.g. when calling `spin_once`[2]
+    /// with this node as an argument), the guard condition can be used to
+    /// interrupt the wait.
+    ///
+    /// [1]: crate::GuardCondition
+    /// [2]: crate::spin_once
+    pub fn create_guard_condition(
+        &mut self,
+        context: &Context,
```

Comment:
Sorry, I have one more comment.

It would be nice if this method
- took an un-boxed callback like `GuardCondition::new()`
- did not require passing in the context. The node already has the context, but only the `rcl_context_t`, not the `Context`. To make this work, the `GuardCondition` could have a non-public method `new_with_rcl_context()` that takes an `rcl_context_t` (or a `Mutex<rcl_context_t>`, you get the idea) . Then `GuardCondition::new()` and `Node::create_guard_condition()` could call `GuardCondition::new_with_rcl_context()`.